source/libraries/ContractDeployer.ts

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env node
+
+import * as binascii from "binascii";
+import * as path from "path";
+import * as EthAbi from "ethjs-abi";
+import * as HttpProvider from "ethjs-provider-http";
+import * as Eth from "ethjs-query";
+import * as EthContract from "ethjs-contract";
+// TODO: Update TS type definition for ContractBlockchainData to allow for empty object (e.g. upload() & uploadAndAddToController())?
+import { ContractBlockchainData, ContractReceipt } from "contract-deployment";
+
+
+export class ContractDeployer {
+    private eth;
+    private compiledContracts;
+    private uploadedContracts;
+    private fromAddress;
+    private gasAmount;
+    private controller;
+
+    public getUploadedContracts() {
+        return this.uploadedContracts;
+    }
+
+    public async initialize(eth: Eth, contractJson: string, fromAddress: string, gasAmount: number): Promise<boolean> {
+        this.eth = eth;
+        this.compiledContracts = JSON.parse(contractJson);
+        this.uploadedContracts = {};
+        this.fromAddress = fromAddress;
+        this.gasAmount = gasAmount;
+
+        this.controller = await this.upload("../source/contracts/Controller.sol");
+        const ownerAddress = (await this.controller.owner())[0];
+        if (ownerAddress != fromAddress) {
+            throw new Error("Controller owner does not equal from address");
+        }
+
+        await this.uploadAllContracts();
+        await this.whitelistTradingContracts();
+        await this.initializeAllContracts();
+        await this.approveCentralAuthority();
+
+        return true;
+    }
+
+    public async uploadAndAddToController(relativeFilePath: string, lookupKey: string = "", signatureKey: string = "", constructorArgs: any = []): Promise<ContractBlockchainData> {
+        lookupKey = (lookupKey == "") ? path.basename(relativeFilePath).split(".")[0] : lookupKey;
+        const contract = await this.upload(relativeFilePath, lookupKey, signatureKey, constructorArgs);
+        if (typeof contract == "undefined") {
+            return {};
```

Comment:
This isn't a valid return type for this function.  Recommend `throw` instead if this is a failure scenario (it looks like it is).  Same comment below.

Reply:
Ok, sounds good.  I'll change this to a throw.